lib/pbench/server/auth/__init__.py

```diff
@@ -53,6 +53,9 @@ def __init__(
         )
         self.verify = verify
         self.connection = requests.session()
+        self.connection.hooks = {
+            "response": lambda r, *args, **kwargs: r.raise_for_status()
+        }
```

Comment:
Why do this when you also added the explicit `raise_for_status` in each of your `_get`|`_put`|`_post` wrappers? The hook is cute, if it works, and might be a good idea, but don't do both!

Reply:
Oops I forgot to remove the ones from the wrapper after some testing

Reply:
It turns out, as the code is presently constructed, the hook is counter-productive, because it prevents the caller from receiving the `response` object, which the exception handling currently uses!  😢